### PR TITLE
deps!: migrate plugins to 0.9 RC

### PR DESCRIPTION
**Pull request by cabljac**
_Before you submit a pull request, please make sure you have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md)._

**This pull request is related to:**

- [ ] A bug
- [ ] A new feature
- [ ] Documentation
- [x] Other: The `next` version of Genkit (0.9) has many breaking changes for the plugin API. This PR migrates some 3P plugins to an RC of `0.9`.

**I have checked the following:**

- [x] I have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md);
- [x] I have added new tests (for bug fixes/features);
- [ ] I have added/updated the documentation (for bug fixes / features).

**Description:**
Migrate several 3P plugins to `0.9`
**Related issues:**
https://github.com/firebase/genkit/issues/1088

### CODE REVIEW

File: examples/anthropic/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "build": "tsc",
+    "build:watch": "tsc --watch",
+    "genkit:dev": "cross-env GENKIT_ENV=dev npm run dev",
+    "dev": "export GENKIT_RUNTIME_ID=$(openssl rand -hex 8) && node lib/index.js 2>&1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "genkit": "^0.9.0-dev.2",
```
Comment (pavelgj) [inline examples/anthropic/package.json:18]:
```suggestion
    "genkit": "^0.9.0-dev.2 | ^0.9",
```

File: plugins/anthropic/src/claude.ts [hunk 2]
```diff
@@ -259,7 +250,9 @@ export function toAnthropicMessageContent(
       extractDataFromBase64Url(part.media.url) ?? {};
     if (!data) {
       throw Error(
-        `Invalid genkit part media provided to toAnthropicMessageContent: ${JSON.stringify(part.media)}.`
+        `Invalid genkit part media provided to toAnthropicMessageContent: ${JSON.stringify(
+          part.media
+        )}.`
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/anthropic/src/claude.ts:255]: Not within scope of PR

File: plugins/anthropic/src/claude.ts [hunk 3]
```diff
@@ -288,12 +281,14 @@ export function toAnthropicMessageContent(
     };
   }
   throw Error(
-    `Unsupported genkit part fields encountered for current message role: ${JSON.stringify(part)}.`
+    `Unsupported genkit part fields encountered for current message role: ${JSON.stringify(
+      part
+    )}.`
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/anthropic/src/claude.ts:286]: Not within scope of PR

File: plugins/cohere/package.json [hunk 4]
```diff
@@ -20,12 +20,10 @@
   "author": "TheFireCo",
   "license": "Apache-2.0",
   "dependencies": {
-    "cohere-ai": "^7.9.5",
-    "zod": "^3.22.4"
+    "cohere-ai": "^7.9.5"
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/cohere/package.json:23]: genkit now re-exports {z} from zod

File: package.json [hunk 5]
```diff
@@ -17,9 +17,15 @@
...
+    "dotenv": "^16.4.5",
+    "genkit": "^0.9.0-dev.2",
     "husky": "^9.0.11",
     "lerna": "^8.1.2",
     "lint-staged": "^15.2.2",
     "prettier": "^3.2.5"
+  },
+  "dependencies": {
+    "uuid": "^11.0.2"
   }
```
[the remark this answers is not in the document]
Comment (cabljac) [reply package.json:30]: This was added in to prevent an error in one of the plugins.

**Review by xavidop — APPROVED** [review]
LGTM

File: plugins/cohere/src/command.ts [hunk 6]
```diff
@@ -174,11 +169,11 @@ export function jsonSchemaToPythonType(schema: Record<string, any>): string {
...
     Object.fromEntries<Cohere.ToolParameterDefinitionsValue>(
-      Object.entries(tool.inputSchema.properties).map(([key, value]) => [
+      Object.entries(tool.inputSchema?.properties).map(([key, value]) => [
         key,
         {
           type: jsonSchemaToPythonType(value as Record<string, any>),
-          required: tool.inputSchema.required?.includes(key) || false,
+          required: tool.inputSchema?.required?.includes(key) || false,
         },
       ])
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/cohere/src/command.ts:178]: This is a fair point, will look into this and resolve

File: plugins/cohere/src/index.ts [hunk 7]
```diff
@@ -14,51 +14,40 @@
...
-);
+
+    for (const name of Object.keys(SUPPORTED_COMMAND_MODELS)) {
+      commandModel(ai, name, client);
+    }
+
+    for (const name of Object.keys(SUPPORTED_EMBEDDING_MODELS)) {
+      cohereEmbedder(ai, name, options);
+    }
+  });
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/cohere/src/index.ts:51]: out of scope

File: plugins/groq/src/groq_models.ts [hunk 8]
```diff
@@ -91,7 +87,7 @@ export const llama3x70b = modelRef({
 
 // Best at JSON mode
 export const mixtral8x7b = modelRef({
-  name: 'groq/mixtral-8x7b-32768',
+  name: 'groq/mixtral-8x7b',
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/groq/src/groq_models.ts:90]: good catch, will investigate

[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/groq/src/groq_models.ts:90]: this potential bug is out of scope of the PR as it exists already in the code

File: plugins/mistral/src/index.ts [hunk 9]
```diff
@@ -14,47 +14,44 @@
...
-      ],
-      embedders: [
-        ...Object.keys(SUPPORTED_EMBEDDING_MODELS).map((name) =>
-          mistralEmbedder(name, client)
-        ),
-      ],
-    };
-  }
-);
+
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/mistral/src/index.ts:46]: out of scope of PR

File: plugins/mistral/src/mistral_llms.ts [hunk 10]
```diff
@@ -105,7 +103,7 @@ export const openMistralSmall = modelRef({
 export const openMistralMedium = modelRef({
   name: 'mistral/mistral-medium-latest',
   info: {
-    versions: ['istral-medium-231'],
+    versions: ['mistral-medium-231'],
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/mistral/src/mistral_llms.ts:106]: out of scope

File: plugins/mistral/src/mistral_llms.ts [hunk 11]
```diff
@@ -257,22 +253,24 @@ export function toMistralRequestBody(
...
     messages: mistralMessages,
     tools: request.tools?.map(toMistralTool),
     model: mappedModelName,
-    max_tokens: request.config?.maxOutputTokens,
+    maxTokens: request.config?.maxOutputTokens,
     temperature: request.config?.temperature,
-    top_p: request.config?.topP,
+    topP: request.config?.topP,
+    // TODO: Mistral API doesn't seem to support the field "n"?
+    // @ts-ignore
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/mistral/src/mistral_llms.ts:269]: Out of scope of this PR; preexisting.

**Review by xavidop — CHANGES_REQUESTED** [review]
the readme of each plugin needs to be updated

File: plugins/openai/src/dalle.ts [hunk 12]
```diff
@@ -58,7 +51,7 @@ function toDallE3Request(
 ): ImageGenerateParams {
   const options = {
     model: 'dall-e-3',
-    prompt: new Message(request.messages[0]).text(),
+    prompt: new Message(request.messages[0]).text,
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/openai/src/dalle.ts:54]: out of scope, azure-openai is not fully implemented

File: plugins/mistral/src/embedders.ts [hunk 13]
```diff
@@ -36,9 +42,7 @@ export function mistralEmbedder(name: string, client: any) {
     async (input, _) => {
       const embeddings = await client.embeddings({
         model: name,
-        input: input.map((d) => {
-          return d.text();
-        }),
+        input: input.map((d) => d.text),
       });
```
[the remark this answers is not in the document]
Comment (cabljac) [reply plugins/mistral/src/embedders.ts:46]: out of scope

File: examples/anthropic/package.json (hunk 1 above)
Comment (pavelgj) [inline examples/anthropic/package.json:18]:
```suggestion
    "genkit": "^0.9.0-dev | ^0.9.0-rc | ^0.9",
```

**Review by pavelgj — APPROVED** [review]

**Review by davidoort — COMMENTED** [review]
I see a bunch of files being deleted (e.g. prompt files, README, .env.local) would be helpful to understand the reasoning behind this. Are they just incompatible or they've been moved somewhere else?

File: examples/cohere/package.json [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    "build": "tsc",
+    "build:watch": "tsc --watch",
+    "genkit:dev": "cross-env GENKIT_ENV=dev npm run dev",
+    "dev": "export GENKIT_RUNTIME_ID=$(openssl rand -hex 8) && node lib/index.js 2>&1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "genkit": "^0.9.0-dev.2",
```
Comment (pavelgj) [inline examples/cohere/package.json:18]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9",
```

File: examples/groq/package.json [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+    "build": "tsc",
+    "build:watch": "tsc --watch",
+    "genkit:dev": "cross-env GENKIT_ENV=dev npm run dev",
+    "dev": "export GENKIT_RUNTIME_ID=$(openssl rand -hex 8) && node lib/index.js 2>&1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "genkit": "^0.9.0-dev.2",
```
Comment (pavelgj) [inline examples/groq/package.json:18]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9",
```

File: examples/mistral/package.json [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+    "build": "tsc",
+    "build:watch": "tsc --watch",
+    "genkit:dev": "cross-env GENKIT_ENV=dev npm run dev",
+    "dev": "export GENKIT_RUNTIME_ID=$(openssl rand -hex 8) && node lib/index.js 2>&1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "genkit": "^0.9.0-dev.2",
```
Comment (pavelgj) [inline examples/mistral/package.json:18]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9",
```

File: examples/openai/package.json [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+    "build": "tsc",
+    "build:watch": "tsc --watch",
+    "genkit:dev": "cross-env GENKIT_ENV=dev npm run dev",
+    "dev": "export GENKIT_RUNTIME_ID=$(openssl rand -hex 8) && node lib/index.js 2>&1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "genkit": "^0.9.0-dev.2",
```
Comment (pavelgj) [inline examples/openai/package.json:18]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9",
```

File: package.json [hunk 18]
```diff
@@ -17,9 +17,15 @@
     "**/*.{js,ts,json,md}": "prettier --write"
   },
   "devDependencies": {
+    "cross-env": "^7.0.3",
+    "dotenv": "^16.4.5",
+    "genkit": "^0.9.0-dev.2",
```
Comment (pavelgj) [inline package.json:22]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9",
```

File: plugins/anthropic/package.json [hunk 19]
```diff
@@ -24,13 +24,11 @@
...
   "author": "TheFireCo",
   "license": "Apache-2.0",
-  "dependencies": {
-    "@anthropic-ai/sdk": "^0.24.0",
-    "zod": "^3.23.8"
-  },
   "peerDependencies": {
-    "@genkit-ai/ai": "^0.5.0",
-    "@genkit-ai/core": "^0.5.0"
+    "genkit": ">=0.9.0-dev.2 <1.0.0 || ^0.9"
```
Comment (pavelgj) [inline plugins/anthropic/package.json:28]:
```suggestion
    "genkit": "~0.9.0-rc || ^0.9"
```

File: plugins/cohere/package.json [hunk 20]
```diff
@@ -20,12 +20,10 @@
...
   },
   "peerDependencies": {
-    "@genkit-ai/ai": "^0.5.0",
-    "@genkit-ai/core": "^0.5.0"
+    "genkit": ">=0.9.0-dev.2 <1.0.0 || ^0.9"
```
Comment (pavelgj) [inline plugins/cohere/package.json:26]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9"
```

File: plugins/groq/package.json [hunk 21]
```diff
@@ -24,16 +24,17 @@
...
   "license": "Apache-2.0",
   "dependencies": {
-    "groq-sdk": "^0.3.3",
-    "zod": "^3.22.4"
+    "groq-sdk": "^0.3.3"
   },
   "peerDependencies": {
-    "@genkit-ai/ai": "^0.5.0",
-    "@genkit-ai/core": "^0.5.0"
+    "genkit": ">=0.9.0-dev.2 <1.0.0 || ^0.9"
```
Comment (pavelgj) [inline plugins/groq/package.json:30]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9"
```

File: plugins/mistral/package.json [hunk 22]
```diff
@@ -20,12 +20,10 @@
...
   "license": "Apache-2.0",
   "dependencies": {
-    "@mistralai/mistralai": "^0.2.0",
-    "zod": "^3.22.4"
+    "@mistralai/mistralai": "^0.2.0"
   },
   "peerDependencies": {
-    "@genkit-ai/ai": "^0.5.0",
-    "@genkit-ai/core": "^0.5.0"
+    "genkit": ">=0.9.0-dev.2 <1.0.0 || ^0.9"
```
Comment (pavelgj) [inline plugins/mistral/package.json:26]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9"
```

File: plugins/openai/package.json [hunk 23]
```diff
@@ -24,12 +24,10 @@
...
   "license": "Apache-2.0",
   "dependencies": {
-    "openai": "^4.47.1",
-    "zod": "^3.23.8"
+    "openai": "^4.47.1"
   },
   "peerDependencies": {
-    "@genkit-ai/ai": "^0.5.0",
-    "@genkit-ai/core": "^0.5.0"
+    "genkit": ">=0.9.0-dev.2 <1.0.0 || ^0.9"
```
Comment (pavelgj) [inline plugins/openai/package.json:30]:
```suggestion
    "genkit": "^0.9.0-rc || ^0.9"
```

**Comment by pavelgj** [comment]
> I see a bunch of files being deleted (e.g. prompt files, README, .env.local) would be helpful to understand the reasoning behind this. Are they just incompatible or they've been moved somewhere else?

It appears the prompt files were unused, so were deleted. More samples were added (per plugin). Perhaps some tool calling samples did not survive the move... @cabljac could you please take a look?

**Comment by cabljac** [comment]
> > I see a bunch of files being deleted (e.g. prompt files, README, .env.local) would be helpful to understand the reasoning behind this. Are they just incompatible or they've been moved somewhere else?
> 
> It appears the prompt files were unused, so were deleted. More samples were added (per plugin). Perhaps some tool calling samples did not survive the move... @cabljac could you please take a look?

Yeah they were unused and the example was fairly out of date, so i replaced it with new samples. I've push a change now to reinstate and update that original example code though :)

**Comment by pavelgj** [comment]
Not to rush, but Genkit 0.9.0 is now released. It would be great if these plugins could be updated as soon as possible.

**Review by davidoort — DISMISSED** [review]

**Comment by EPMatt** [comment]
Hey @cabljac ,
thank you for your contribution. I've fixed build issues related to this PR - plus bumped the genkit version to `v0.9.0` stable.
As I don't have push access to this branch, I've uploaded the edits to a separate branch: https://github.com/TheFireCo/genkit-plugins/tree/pr/cabljac/155 and opened a PR on your fork.

**Pull request closed without merging**